spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -470,6 +470,10 @@ class SpdxLicenseWithExceptionExpression(
                     )
                 }
             }
+            else -> throw SpdxException(
```

Comment:
Nit: Blank line before this one to align with spacing above.

Reply:
And can we even reach this? Because shouldn't in this case the `as` cast in line 455 fail before?

Reply:
> And can we even reach this? Because shouldn't in this case the as cast in line 455 fail before?

`SpdxSingleLicenseExpression` can be:
- `SpdxLicenseIdExpression`
- `SpdxLicenseReferenceExpression`
- `SpdxLicenseWithExceptionExpression`
- `SpdxSimpleExpression`

From what I see this would cover everything, so the else branch would not be required.
However: IntelliJ complains that it is required and marks it as an error.

How should I change this? Replace the `when` with an if / else? :/

Reply:
I would have said to suppress the inspection hint, but there is [contradicting information whether this is possible](https://stackoverflow.com/questions/61756519/how-can-i-disable-the-non-exhaustive-when-expression-inspection). So, can't we just ignore this like we did until now?

Reply:
I did not find a way to disable it. I'll remove the commit.

Reply:
I noticed that after a recent update IntelliJ started complaining about this but it was fine when I wrote this code. I was hoping that it would get fixed in another update of the Kotlin plugin but didn't check if there is a bug report for this. We had another case recently where IntelliJ wrongly reported a compiler error, but I don't remember where this was right now. This is getting a bit annoying.